cmake/onnxruntime_mlas.cmake

```diff
@@ -124,6 +124,15 @@ if(MSVC)
     )
   endif()
 else()
+  if (CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
```

Comment:
dumpmachine sometimes doesn't work well in my machine, reporting some linker issue.
This change looks more robust.